Documents/AnalyticsToolDesignDocument.md

```diff
@@ -0,0 +1,48 @@
+# VFS For Git: Analytics Tooling Design Document
+
+This document details the design of and plan for the **VFS4G analytics tool**. The tool aims to help users identify reasons why their enlistment may be slowing down and the `gvfs` commands running slowly. By collecting and reporting statistics, the power to uncover and hopefully fix problems as they arrive moves to the user, empowering them to use the software to its fullest potential.
+
+## Problem
+
+When users unintentionally hydrate too many files within their enlistment, commands like checkout and commit increase their running time by several orders of magnitude. This causes unacceptable losses in performance and greatly hinders users as they go about their daily development cycle. The primary metric of the overhydration is the number of placeholder files which have been pulled from the remote into the local system. Slowdowns may also stem from having many modified files or directories which increases the amount of work that git does under the hood. Additionally, if a local enlistment has been in use for a long time, the number of loose object may eventually reach the order of several thousand in which case that also contributes to the loss in performance.
+
+## Provided Statistics
+
+### Safe to access
+
+- **Modified paths** - The number of files which have been edited and have been handed over to git
+- **Git index** - The differences between modified paths and the index tracked by git
+- **Placeholders** - A total number of the placeholders being tracked in VFS4G and a list of the placeholders in the system (Note that this contains some upper level directories and thus not all of their children are included in the count without enumeration)

```

Comment:
> Note that this contains some upper level directories and thus not all of their children are included in the count without enumeration

Two small things to be aware of:

- The placeholders collection will always contain *all* placeholders, even those files whose parent folders are in the placeholder list.

- The "Modified paths" can also contain top level directories, and it has the behavior you mentioned here.  If a directory is in modified paths its children will not be included.

Reply:
I updated that for correctness, thank you!